tests/python/gaia-ui-tests/gaiatest/apps/messages/regions/new_message.py

```diff
@@ -27,7 +27,7 @@ class NewMessage(Messages):
 
     def __init__(self, marionette):
         Base.__init__(self, marionette)
-        Wait(self.marionette).until(lambda m: self.apps.displayed_app.name == self.name)
+        Wait(self.marionette).until(lambda m: self.apps.displayed_app.manifest_url == self.manifest_url)
```

Comment:
Yeah, I agree. There are some cases where there is a check for an app not to be displayed, but those are rare and I hope they can be removed at some point.